components/mdns/mdns.c

```diff
@@ -25,6 +25,11 @@
 #include "esp_wifi.h"
 #endif
 
+#if ESP_IDF_VERSION <= ESP_IDF_VERSION_VAL(5, 1, 0)
+#ifndef CONFIG_ESP_WIFI_ENABLED
+#define CONFIG_ESP_WIFI_ENABLED 1
+#endif
+#endif
 
```

Comment:
Suggest that we create a local config called something like `MDNS_ESP_WIFI_ENABLED` and do something like:
```suggestion
#if ESP_IDF_VERSION <= ESP_IDF_VERSION_VAL(5, 1, 0)
  // IDF <= v5.1 does not support enabling/disabling esp-wifi
  #define MDNS_ESP_WIFI_ENABLED 1
#else
  #define MDNS_ESP_WIFI_ENABLED CONFIG_ESP_WIFI_ENABLED
#endif
```

Reply:
I don't think it's wise to update skdconfig option, even if it's not supposed to be defined